### DISCUSSION TITLE
Place and resize image using page and layers (or a better option?)

### DISCUSSION THREAD

**Discussion by BillRaymond** [ImageMagick/ImageMagick#5382] [Help] [answered]
I am using ImageMagick 7.1.x/Alpine Linux to create a featured image for my podcast and am working on a new design, which is the context of this discussion. The featured image will start with a template, then add guest images (faces), text, and other imagery (like a play icon).

**Note**: Scroll to the bottom of this message for mockups I created in Photoshop.

## 🏁 My starting template
![sc-template](https://user-images.githubusercontent.com/15895828/181294777-017ffe21-3f57-4ce7-8bb7-8e095e96e4c9.png)

I basically break the featured image into 3 columns:

* Column 1 is in the default content already in the template, which contains the logo and my face
* Column 2 is dynamically managed by ImageMagick and where all the guest images go. The more guests, the smaller the faces. The column should not exceed 282 pixels wide
  * 1 guest: 282x282 pixels, placed in the vertical center of the image
  * 2 guests: 282x282 pixels, spaced evenly within the column, centered vertically to the overall template height
  * 3 guests: 200x200 pixels, spaced evenly within the column, centered vertically to the overall template height
  * 4 guests: 150x150 pixels, spaced evenly within the column, centered vertically to the overall template height
* Column 3 is dynamically managed by ImageMagick and where I place various text elements, a play icon, and perhaps some other elements

![columns](https://user-images.githubusercontent.com/15895828/181308618-50f6af6f-9218-40f4-bd2c-fa8fdd5adc0b.png)

## 🤞 Requirement
I want all my code to create the featured image in one ImageMagick `convert` line using ImageMagick 7.1.x for Linux (Alpine, ASH).

## 🧑‍💻 My current code
Right now, I have four `convert` code blocks. The first one places the single guest. The second places two guests, and so on. Since one or two guests are 282x282 pixels, it is easy to place them with ImageMagic.

Where things start to get messy is when I try to add three or more guests, because not only do I need to place the guest in a specific location, but it needs to be resized.

Here is the code that builds the 3-up version:
```
###  Featured image for three guests
magick convert sc-template.png `# load template background image`\
    -page +972+448 sc-play.png `# load play icon image`\
    -layers flatten `# combine layers`\
    -page +256+41 face.png `# load guest 1 image`\
    -page +256+246 face2.png `# load guest 2 image`\
    -page +256+451 face3.png `# load guest 2 image`\
    -layers flatten `# combine layers`\
    -font './share-card-creator/fonts/ProximaNovaA-Bold.ttf' `# load the font`\
    -fill '#f49f1c' -background none `# The podcast title does not have a background`\
    -size 580x40 caption:'PODCAST' `# Add the word PODCAST to the template`\
    -geometry +550+46 `# Set the x and y position for the PODCAST text`\
    -composite `# Add text to the image`\
    -fill white -background none `# The title for the podcast does not have a background`\
    -size 580x340 caption:'Outcome-driven design and team behaviors' `# Podcast title as it appears on the website`\
    -geometry +550+96 `# Set the x and y location for the podcast title`\
    -composite `# Add the podcast's title to the image`\
    -fill white -background none `# the guest names do not have a background`\
    -size 405x195 caption:'with A Cool Person and Another Cool Person' `# List the guest names for the podcast`\
    -geometry +550+446 `# Set the x and y position of the guest names`\
    -composite `# Add guest names to the image`\
    sc.png
exit
```

### Result with existing code
After I run the code, of course, the images are too large and they overlap, like this: 
![issue](https://user-images.githubusercontent.com/15895828/181303035-768c1820-6e36-4cbc-a672-bc97b018d3f8.png)

## ⚠️ Problem
The problem I am having is I can write the `page` code so it places the image in the right location, but I cannot seem to find a way to resize it. I tried this:
```
    -page 200x200+256+451 face3.png `# load guest 2 image`\
```
And I tried many other permutations of that (putting the size in parenthesis, etc) but I cannot seem to get it to resize, just place the image based on geometry.

## ❓Question
I guess I have two questions.

1. Can I both resize and place the image using `-page` and if so, how? The documentation suggests you can but I cannot find a good example (perhaps I am searching wrong).
2. Is there a better way to approach this? I do not want to use `montage` because my requirement is the code is all in one line. But perhaps there is a way with `convert` that I can say "here is a rectangle, now place the images in it, centering them vertically and resizing them so they don't overlap"?

Note that at the moment, I assume there will never be more than four guests on a podcast, which is why I did not design for more.

Thank you in advance!

## Examples of all four scenarios (Photoshop markup)
![1-up](https://user-images.githubusercontent.com/15895828/181301149-f49d9d9a-faac-4beb-8953-e646c8af83af.png)
![2-up](https://user-images.githubusercontent.com/15895828/181301154-ba9885a6-46c9-4b25-bd9b-37281693320f.png)
![3-up](https://user-images.githubusercontent.com/15895828/181301155-626d272c-3670-4f58-9e1f-e6690273f214.png)
![4-up](https://user-images.githubusercontent.com/15895828/181301156-c0c8b15d-3545-4d31-99c3-50b9a805bd6c.png)

**Comment by fmw42** [comment, marked as answer]
In place of 

`-page +256+41 face.png`

You could do

`\( face.png -resize 200x200 -set page +256+41  \) `

etc for face2.png and face3.png

**Comment by BillRaymond** [comment]
That did the trick! Thank you!
